### PR TITLE
Feature/BNGP-3443&3169: Enable upload of proof of ownership for multiple landowners

### PR DESCRIPTION
**Pull request by usman-j-shaikh**

### CODE REVIEW

File: packages/webapp/src/routes/land/check-ownership-proof-file.js
Comment (paulsimonandrews) [inline packages/webapp/src/routes/land/check-ownership-proof-file.js:33]: The filename is already available in `context` so no need to extract it again

File: packages/webapp/src/routes/land/check-ownership-proof-file.js
Comment (paulsimonandrews) [inline packages/webapp/src/routes/land/check-ownership-proof-file.js:26]: Not sure why you are doing this (remove operation followed by setting the redis key) as the file won't have been added to `LAND_OWNERSHIP_PROOFS` at this stage.

Also, even though I don't think you need this line of code, I wouldn't use lodash here, it's an unnecessary dependency for such as simple operation that can be achieved with the built in Array methods (filter or splice with findIndex)

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Some comments in the code to address.

Also, the E2E journey is now broken as the other landowners and consent pages have been bypassed but they are still expected in the payload generated on submission. The test data works, but if you manually fill in the journey it won't work. So:
* Remove the other landowner and consent pages and redis cache variables
* Update the payload (`application.js` and `application-validation`) to remove `landownerConsent` and `otherLandowners`
* Remove seed test data for other landowners and consent

There is also a bug: press back link on "land-ownership-list" page and click "Save and continue" again and it will add the file again on the "land-ownership-list" page. Check the code for uploading multiple files in the legal agreement section of the journey as it doesn't do this.

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Two small changes please (see comments in code). Will approve when done.

**Review by paulsimonandrews — DISMISSED** [review]

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
In additional to other review comments:

-  incorrect copy is displayed when being asked to confirm removal of a document showing proof of land ownership, Save and Continue is pressed and no option has been selected.

image

- If all proof of land ownership files are removed I think the check your answers page should revert to an in progress status and any attempt to navigate to /land/ownership-proof-list with no uploads should redirect to /land/upload-ownership-proof (similar to how the legal agreement section works). At the moment, when the page is viewed in this way it has incorrect pluralisation.

image

- Check and submit page handling for multiple proof of land ownership documents differs from that for multiple legal agreement documents. As I cannot find any guidance in the prototype or copydeck I suggest the approach used for legal agreement documents needs to be followed:

image

image

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
@AjinkyaJawalkar  - The check and submit page is still showing a link for each proof of ownership upload rather than grouping them with one link (as for legal agreement files - see the screenshot on my previous comment). It works but it's inconsistent and the copydeck screenshots only have an example with one file.

@shaundefra has confirmed a defect will be raised if this is not resolved before pull request approval.

**Review by pwadmore-ea — APPROVED** [review]
> @AjinkyaJawalkar - The check and submit page is still showing a link for each proof of ownership upload rather than grouping them with one link (as for legal agreement files - see the screenshot on my previous comment). It works but it's inconsistent and the copydeck screenshots only have an example with one file.
> 
> @shaundefra has confirmed a defect will be raised if this is not resolved before pull request approval.

@shaundefra  and @paulsimonandrews  - Following agreement that the check and submit inconsistency will be treated as a known issue so the next deployment is not delayed I will  approve the pull request.,

**Comment by LeenaKu** [comment]
Tested on POC3 and known issues covered by https://eaflood.atlassian.net/browse/BNGP-3443?focusedCommentId=554208.

**Pull request merged**
